pol-core/pol/syshookscript.cpp

```diff
@@ -604,5 +466,58 @@ size_t ExportScript::estimateSize() const
 {
   return sd.estimatedSize() + uoexec.sizeEstimate();
 }
+
+bool ExportScript::check_bool()
+{
+  if ( uoexec.error() || uoexec.ValueStack.empty() )
+    return false;
+  bool istrue = uoexec.ValueStack.back().get()->isTrue();
+  uoexec.ValueStack.pop_back();
+  return istrue;
+}
+
+int ExportScript::check_int()
+{
+  if ( uoexec.error() || uoexec.ValueStack.empty() )
+    return 0;
+  int ret;
+  BObjectImp* imp = uoexec.ValueStack.back().get()->impptr();
+  if ( imp->isa( BObjectImp::OTLong ) )
+  {
+    BLong* pLong = static_cast<BLong*>( imp );
+    ret = pLong->value();
+  }
+  else if ( imp->isa( BObjectImp::OTBoolean ) )
+  {
+    ret = imp->isTrue() ? 1 : 0;
+  }
+  else
+  {
+    ret = 0;
+  }
+  uoexec.ValueStack.pop_back();
+  return ret;
+}
+
+std::string ExportScript::check_string()
+{
+  if ( uoexec.error() || uoexec.ValueStack.empty() )
+    return "error";
+  auto ret = uoexec.ValueStack.back().get()->impptr()->getStringRep();
+  uoexec.ValueStack.pop_back();
+  return ret;
+}
+
+BObjectImp* ExportScript::check_imp()
```

Comment:
I wonder if this might leak at the call sites, in case something goes wrong between calling the method and returning the BObject for it

Reply:
There is hope :)